### PR TITLE
[cloud-provider-azure] fixed the behavior of the acceleratedNetworking variable

### PR DESCRIPTION
**Pull request by xeonic-ant**
## Description

Fixed helm to render correctly booleans.

## Why do we need it, and what problem does it solve?

Closes https://github.com/deckhouse/deckhouse/issues/4988

## Checklist
- [x] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: cloud-provider-azure
type: fix
summary:  fixed the behavior of the acceleratedNetworking variable 
impact_level: default
```

### CODE REVIEW

**Comment by alexey-gavrilov-flant** [comment]
https://github.com/deckhouse/deckhouse/issues/4988

**Comment by alexey-gavrilov-flant** [comment]
https://github.com/deckhouse/deckhouse/pull/5042

File: modules/040-node-manager/template_tests/module_test.go [hunk 1]
```diff
@@ -1356,6 +1384,31 @@ ccc: ddd
 				assertValues(mcls, "spec.tags")
 			})
 
+			It("spec.properties.networkProfile.acceleratedNetworking default true", func() {
+				Expect(f.RenderError).ShouldNot(HaveOccurred())
+
+				t := f.KubernetesResource("AzureMachineClass", "d8-cloud-instance-manager", "worker-02320933")
+				Expect(t.Exists()).To(BeTrue())
+				//checksum := mcdt.Field("spec.properties.networkProfile.acceleratedNetworking").String()
```
Comment (alexey-gavrilov-flant) [inline modules/040-node-manager/template_tests/module_test.go:1392]: You have to delete the line with the comment

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @alexey-gavrilov-flant. I've started the workflow for label 'e2e/run/azure'...

File: modules/040-node-manager/template_tests/module_test.go [hunk 2]
```diff
@@ -1356,6 +1384,30 @@ ccc: ddd
 				assertValues(mcls, "spec.tags")
 			})
 
+			It("spec.properties.networkProfile.acceleratedNetworking default true", func() {
```
Comment (RomanenkoDenys) [inline modules/040-node-manager/template_tests/module_test.go:1387]:
```suggestion
			It("spec.properties.networkProfile.acceleratedNetworking is not ser (default true)", func() {
```

File: modules/040-node-manager/template_tests/module_test.go [hunk 3]
```diff
@@ -1356,6 +1384,30 @@ ccc: ddd
...
+				Expect(f.RenderError).ShouldNot(HaveOccurred())
+
+				t := f.KubernetesResource("AzureMachineClass", "d8-cloud-instance-manager", "worker-02320933")
+				Expect(t.Exists()).To(BeTrue())
+				Expect(t.Field("spec.properties.networkProfile.acceleratedNetworking").Bool()).To(Equal(true))
+			})
+
+			It("spec.properties.networkProfile.acceleratedNetworking true", func() {
```
Comment (RomanenkoDenys) [inline modules/040-node-manager/template_tests/module_test.go:1395]:
```suggestion
			It("spec.properties.networkProfile.acceleratedNetworking is set to true", func() {
```

File: modules/040-node-manager/template_tests/module_test.go [hunk 4]
```diff
@@ -1356,6 +1384,30 @@ ccc: ddd
...
+				Expect(f.RenderError).ShouldNot(HaveOccurred())
+
+				t := f.KubernetesResource("AzureMachineClass", "d8-cloud-instance-manager", "bbb-02320933")
+				Expect(t.Exists()).To(BeTrue())
+				Expect(t.Field("spec.properties.networkProfile.acceleratedNetworking").Bool()).To(Equal(true))
+			})
+
+			It("spec.properties.networkProfile.acceleratedNetworking false", func() {
```
Comment (RomanenkoDenys) [inline modules/040-node-manager/template_tests/module_test.go:1403]:
```suggestion
			It("spec.properties.networkProfile.acceleratedNetworking is set to false", func() {
```

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
some nits

**Comment by miklezzzz** [comment]
@xeonic-ant Hi, review suggestions please. For now, the PR isn't getting into 1.47.x.

**Comment by z9r5** [comment]
Please fill the changes section accroding to changes.

**Pull request merged**